### ISSUE TITLE
Widget Wrong Casing for the Props Leads to Error

### ISSUE THREAD

**Issue by aress31** [rjsf-team/react-jsonschema-form#4407]
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs)
- [ ] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net), [Codesandbox.io](https://codesandbox.io) or preferably a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### What theme are you using?

mui

### Version

5.23.1

### Current Behavior

The properties passed down to customer widget/fields have the wrong casing causing both lighthouse errors and them not being applied, see:

```js
const createFields = () => ({
  ArrayField: AutocompleteArrayField,
});

const AutocompleteArrayField = ({
  formData,
  schema: { title },
  onChange,
  ...props
}) => {
  const handleChange = (event, newValue) => onChange(newValue);

  return (
    <Autocomplete
      freeSolo
      multiple
      options={[]}
      value={formData}
      onChange={handleChange}
      renderInput={(params) => <TextField {...params} label={title} />}
      {...props}
    />
  );
};
```

![image](https://github.com/user-attachments/assets/3832bb8b-81f7-47d2-bab6-09ab10606db1)

### Environment

```markdown
- OS:
- Node:
- npm:
```

**Comment by nickgros** [comment]
Good catch, these props should be pulled out before we pass them along. @aress31 , would you be interested in creating a PR to fix this in the widget(s) where you notice the issue?

**Comment by aress31** [comment]
They should not all be filtered out, just passed properly, for example the `required`, `autofocus`, `autocomplete` are needed, just in the proper React casing.

Also, I am not available to work on a PR.

**Comment by heath-freenome** [comment]
@aress31 Since you are spreading `...props` onto the Mui `Autocomplete` yourself, you will have to transform them from the props that RJSF uses to ones that `Autocomplete` uses yourself. This is not a bug
